src/sagemaker/spark/processing.py

```diff
@@ -940,9 +940,18 @@ def _extend_processing_args(self, inputs, outputs, **kwargs):
             outputs: Processing outputs.
             kwargs: Additional keyword arguments passed to `super()`.
         """
+
+        if inputs is None:
+            inputs = []
+
+        # make a shallow copy of user inputs
```

Comment:
nit: can use the library - https://docs.python.org/3/library/copy.html in future.

Reply:
+1 on using this library as the shallow copy code block repeats in multiple places